pkg/ingester/metrics.go

```diff
@@ -216,6 +216,17 @@ func newIngesterMetrics(r prometheus.Registerer, createMetricsConflictingWithTSD
 	return m
 }
 
+func (m *ingesterMetrics) deletePerUserMetrics(userID string) {
+	m.memMetadataCreatedTotal.DeleteLabelValues(userID)
+	m.memMetadataRemovedTotal.DeleteLabelValues(userID)
+	m.activeSeriesPerUser.DeleteLabelValues(userID)
+
+	if m.memSeriesCreatedTotal != nil {
+		m.memSeriesCreatedTotal.DeleteLabelValues(userID)
+		m.memSeriesRemovedTotal.DeleteLabelValues(userID)
```

Comment:
I know the current code is correct, but just for paranoia (against possible future changes) and clarity I would add an extra `if m.memSeriesRemovedTotal != nil`